### ISSUE TITLE
Improving our Label System

### ISSUE THREAD

**Issue by dignifiedquire** [filecoin-project/go-filecoin#720]
While we have some labels on issues, I would like to standardize them, and make it easy
for everyone on the project to glance important information about issues from them.

This proposal is based on the [issue conventions from rust-lang/rust](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#issue-triage).

## Issue Categories

- Yellow, A-prefixed labels state which area of the project an issue relates to.
  - `A-consensus`
  - `A-mining`
  - `A-api`
  - `A-crypto`
  - `A-wallet`
  - `A-commands`
  - `A-chain`
  - `A-vm`
  - `A-address`
  - `A-testing`
  - `A-config`
  - `A-lookup`
  - `A-proofs`
  - `A-state`
  - `A-node`
- Dark blue, `testnet` label track changes which need to be implemented for testnet.
- Light purple, C-prefixed labels represent the category of an issue.
  - `C-bug`: Category: This is a bug.
  - `C-cleanup`: Category: PRs that clean code up or issues documenting cleanup.
  - `C-enhancement`: Category: A PR with an enhancement or a proposed on in an issue.
  - `C-feature-accepted`: Category: A feature request that has been accepted pending implementation.
  - `C-feature-request`: Category: A feature request, i.e: not implemented / a PR.
- Green, E-prefixed labels explain the level of experience necessary to fix the issue.
  - `E-easy`: Call for participation: Experience needed to fix: Easy / not much
  - `E-hard`: Call for participation: Experience needed to fix: Hard / a lot
  - `E-medium`: Call for participation: Experience needed to fix: Medium / intermediate
  - `E-help-wanted`: Call for participation: Help is requested / needed by the one who filed the issue to fix it.
  - `E-mentor`: Call for participation: This issue is currently mentored.
  - `E-needs-mentor`: Call for participation: This issue is in need of a mentor.
  - `E-needstest`: Call for participation: writing correctness tests
- The dark blue `needs-design-doc` are issues that require more detailed specification before they can be implemented.
- Red, I-prefixed labels indicate the importance of the issue.
  - `I-needs-decision`: Issues in need of a decision.
  - `I-security :bomb:`: Issues with critical security implications.
  - `I-panic`: Issues that cause program panics.
- The purple metabug label marks lists of bugs collected by other categories.
  - `metabug`: Issues about issues themselves ("bugs about bugs")
- Purple gray, O-prefixed labels are the operating system or platform that this issue is specific to.
  - `O-linux`: Operating System: Linux
  - `O-macos`: Operating System: MacOS
  - `O-windows`: Operating System: Windows
- Orange, P-prefixed labels indicate a bug's priority. These labels are only assigned during triage meetings, and replace the I-nominated label.
  - `P0` - needs to be addressed right now, stop what you are doing and pick this up
  - `P1` - high priority, we need to get to this next with probability 1
  - `P2` - we should get to this, eventually
  - `P3` - not a priority
- Pink, `regression` label tracks regressions.
- Gray, S-prefixed labels are used for tracking the status of pull requests.
  - `S-blocked-closed`: Status: marked as blocked ❌ and closed because resolving the block is expected to take a long time.
  - `S-blocked`: Status: marked as blocked ❌ on something else such as a design doc or other implementation work.
  - `S-inactive-closed`: Status: closed due to inactivity.
  - `S-waiting-on-author`: Status: The marked PR is awaiting some action (such as code changes) from the PR author.
  - `S-waiting-on-bikeshed`: Status: Awaiting a decision on trivial things.
  - `S-waiting-on-perf`: Status: Waiting on a perf run to be completed.
  - `S-waiting-on-review`: Status: Awaiting review from the assignee but also interested parties.
- Light green, WG-prefixed labels indicate which working group is responsible and/or needed for this
  - `WG-go-filecoin`: Working Group: go-filecoin-wg
  - `WG-proofs`: Working Group: proofs-wg
  - `WG-crucible`: Working Group: crucible-wg
- Misc other labels
  - `invalid`: This is not an actual issue
  - `Epic`: Indicates a epic main issue (used & set by zenhub)
  - `ipfs`: Related to an issue in IPFS land

## Next Steps

I will collect feedback on this for the next 24h and pending any big blockers move forward with this. Labels are easy enough to change if there are issues with it.
This means doing the two following things

- [ ] adding the above documentation to `CONTRIBUTING.md`
- [ ] add the suggest labels to the go-filecoin & rust-proofs repo

**Comment by mishmosh** [comment]
Closing for now because system is working reasonably well, reopen as needed (or create new issue with new problem statement).
